### PR TITLE
Cross namespace instrumentation

### LINKED ISSUE #886: support cross-namespace instrumentation references in annotations
**Is your feature request related to a problem? Please describe.**
Currently, it is required to define an `Instrumentation` object in every namespace in which you wish to use the operator to auto-instrument pods. These many `Instrumentation`s may represent duplicate configurations.

**Describe the solution you'd like**
Being able to define an `Instrumentation` in one namespace and reference it in the annotations of pods in other namespaces would allow for better centralisation of `Instrumentation` configuration when required.

I.e.
```yaml
instrumentation.opentelemetry.io/inject-java: "tracing-components/my-instrumentation"
```

**Describe alternatives you've considered**
Alternatives to this would be using a helm chart or similar to approach to replicate a single configuration across multiple namespaces, or using another operator to replicate `Instrumentation` resources in a similar way to the many secret/configmap synchronisation solutions.

### PR DESCRIPTION
**Pull request by tKe**
This PR adds support for specifying the namespace of an instrumentation instance in the `inject-*` annotations.

closes #886

### CODE REVIEW

**Comment by ringerc** [comment]
See also https://github.com/open-telemetry/opentelemetry-operator/issues/886#issuecomment-1139214209

**Review by VineethReddy02 — APPROVED** [review]
Thanks for this PR! @tKe 
LGTM!

@pavolloffay can you review this PR from your end? :)

**Review by yuriolisa — APPROVED** [review]
LGTM!

File: pkg/instrumentation/podmutator.go [hunk 1]
```diff
@@ -118,8 +118,15 @@ func (pm *instPodMutator) getInstrumentationInstance(ctx context.Context, ns cor
 		return pm.selectInstrumentationInstanceFromNamespace(ctx, ns)
 	}
 
+	var instNamespacedName types.NamespacedName
```
[the remark this answers is not in the document]
Comment (tKe) [reply pkg/instrumentation/podmutator.go:121]: This pod mutator is only used for instrumentation. The sidecar injection uses a separate pod mutator which is not impacted. The sidecar injection cannot currently work across namespaces due to the requirement to mount the collector configmap.

**Comment by ringerc** [comment]
I love PRs where the `README.md` update is right there when it is opened.

This is an exciting change to have. I was about to hack up my infrastructure to add a complicated mess of Kustomize `Component` injections to work around this - and now I don't have to.

**Pull request merged**